crates/hyperswitch_domain_models/src/router_request_types.rs

```diff
@@ -99,6 +112,38 @@ pub struct ConnectorCustomerData {
     pub payment_method_data: PaymentMethodData,
 }
 
+impl TryFrom<SetupMandateRequestData> for ConnectorCustomerData {
+    type Error = error_stack::Report<ApiErrorResponse>;
+    fn try_from(data: SetupMandateRequestData) -> Result<Self, Self::Error> {
+        Ok(Self {
+            email: data.email,
+            payment_method_data: data.payment_method_data,
+            description: None,
+            phone: None,
+            name: None,
+            preprocessing_id: None,
+        })
+    }
+}
+impl<F> TryFrom<&RouterData<F, PaymentsAuthorizeData, response_types::PaymentsResponseData>>
```

Comment:
can we have specialized type here instead of a generic?

Reply:
Not currently. Because the aliases for RouterData are currently defined in router crate.

Reply:
The Generic F can be a  specialized type right, in this case it will be `api::Authorize`